### PR TITLE
Optimizer typing

### PR DESCRIPTION
**Pull request by autoih**
Linked to #894.

### CODE REVIEW

File: tensorflow_addons/optimizers/novograd.py [hunk 1]
```diff
@@ -15,9 +15,12 @@
 """NovoGrad for TensorFlow."""
 
 import tensorflow as tf
+from tensorflow_addons.utils.types import FloatTensorLike
 
 # TODO: Find public API alternatives to these
 from tensorflow.python.training import training_ops
+from typing import Union, Callable
+from typeguard import typechecked
```
Comment (seanpmorgan) [inline tensorflow_addons/optimizers/novograd.py]: Move these above the TODO please

File: tensorflow_addons/optimizers/rectified_adam.py [hunk 2]
```diff
@@ -65,19 +69,20 @@ class RectifiedAdam(tf.keras.optimizers.Optimizer):
...
-        min_lr=0.0,
-        name="RectifiedAdam",
+        learning_rate: Union[FloatTensorLike, Callable] = 0.001,
+        beta_1: FloatTensorLike = 0.9,
+        beta_2: FloatTensorLike = 0.999,
+        epsilon: FloatTensorLike = 1e-7,
+        weight_decay: FloatTensorLike = 0.0,
+        amsgrad: bool = False,
+        sma_threshold: FloatTensorLike = 5.0,
+        total_steps: FloatTensorLike = 0,
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/optimizers/rectified_adam.py]:
```suggestion
        total_steps: int = 0,
```

Comment (autoih) [reply tensorflow_addons/optimizers/rectified_adam.py]: Thanks @gabrieldemarmiesse.

File: tensorflow_addons/optimizers/novograd.py (hunk 1 above)
Comment (autoih) [reply tensorflow_addons/optimizers/novograd.py]: Thanks @seanpmorgan.

**Comment by gabrieldemarmiesse** [comment]
Thanks!

**Pull request merged**